rust/libnewsboat/src/utils.rs

```diff
@@ -265,8 +267,78 @@ pub fn unescape_url(rs_str: String) -> Option<String> {
     Some(result.unwrap().replace("\0",""))
 }
 
+pub fn get_command_output(cmd: &str) -> String {
+    let cmd = Command::new("sh").arg("-c").arg(cmd).output();
+    // from_utf8_lossy will convert any bad bytes to U+FFFD
+    cmd.map(|cmd| String::from_utf8_lossy(&cmd.stdout).into_owned())
+        .unwrap_or_else(|_| String::from(""))
+}
+
+pub fn run_command(cmd: &str, param: &str) {
+    let child = Command::new(cmd)
+        .arg(param)
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn();
+    if let Err(error) = child {
+        log!(Level::Debug,
+             &format!("utils::run_command: spawning a child for \"{}\" failed: {}", cmd, error));
+    }
+
+    /* We deliberately *don't* wait for the child to finish. */
+}
+
+pub fn run_program(cmd_with_args: &[&str], input: &str) -> String {
+    if cmd_with_args.is_empty() {
+        return String::new();
+    }
+
+    let child = Command::new(cmd_with_args[0])
+        .args(&cmd_with_args[1..])
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::null())
+        .spawn();
+
+    match child {
+        Err(error) =>
+            log!(Level::Debug,
+                 &format!("utils::run_program: spawning a child for \"{:?}\" \
+                           with input \"{}\" failed: {}",
+                           cmd_with_args,
+                           input,
+                           error)),
+
+        Ok(mut child) => {
+            {
+                if let Some(mut stdin) = child.stdin.as_mut() {
```

Comment:
The `mut` is not necessary. Funny enough it doesn't get caught by the compiler here, but on my review branch it did after using Result combinators.